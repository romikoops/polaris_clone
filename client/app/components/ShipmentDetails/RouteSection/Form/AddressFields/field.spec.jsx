import * as React from 'react'
import { shallow } from 'enzyme'
import Field from './field'

const baseProps = {
  className: 'TestClassName',
  label: 'TestLabel',
  name: 'TestName',
  onBlur: () => {},
  validations: {},
  value: '1234'
}

describe('<Field>', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = { ...baseProps }
    wrapper = shallow(<Field {...props} />)
  })

  it('renders correctly ', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('context disabled', () => {
    beforeEach(() => {
      props = { ...baseProps, disabled: true }
      wrapper = shallow(<Field {...props} />)
    })

    it('renders correctly ', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
