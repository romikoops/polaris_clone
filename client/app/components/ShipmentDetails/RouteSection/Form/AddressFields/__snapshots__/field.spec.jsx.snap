// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field> context disabled renders correctly  1`] = `
<div
  className="field TestClassName"
>
  <label
    htmlFor="TestName"
  >
    TestLabel
  </label>
  <Formsy(FormsyInput)
    disabled={true}
    innerRef={[Function]}
    name="TestName"
    onBlur={[Function]}
    placeholder="TestLabel"
    required={false}
    validationError=""
    validationErrors={Object {}}
    validations={Object {}}
    value="1234"
  />
</div>
`;

exports[`<Field> renders correctly  1`] = `
<div
  className="field TestClassName"
>
  <label
    htmlFor="TestName"
  >
    TestLabel
  </label>
  <Formsy(FormsyInput)
    disabled={false}
    innerRef={[Function]}
    name="TestName"
    onBlur={[Function]}
    placeholder="TestLabel"
    required={false}
    validationError=""
    validationErrors={Object {}}
    validations={Object {}}
    value="1234"
  />
</div>
`;
